### ISSUE TITLE
Compositor Conversion Plus Clustered-Shading Lighting

### ISSUE THREAD

**Issue by Megaf** [c172p-team/c172p#1349]
As the title says.

Use Compositor enabled FlightGear
Disable ALS
Enable Model shader.
When Lights blink, they cast shadows.

![fgfs-20210104181720](https://user-images.githubusercontent.com/6201512/103566056-434d7c80-4eb9-11eb-9655-e3fba48fd69b.png)
![fgfs-20210104181725](https://user-images.githubusercontent.com/6201512/103566028-3b8dd800-4eb9-11eb-9999-79b1cb61a1f7.png)

**Comment by wlbragg** [comment]
Interesting, I am currently getting ready to implement compositor lighting and will take a look at this while I am at it. Thanks for the heads up!

**Comment by wkitty42** [comment]
i've seen this when working with the AI craft's lights... the fix is adding `<noshadow />` in the proper place... for ease, i chose to put it where the lights are grouped for shared animations... here's the block of relatively common code they use on my setup...
```
  <!-- Group Lights for Shared Animations -->
  <animation>
    <name>BeaconGrouped</name>
    <noshadow />
    <object-name>Top_Beacon</object-name>
    <object-name>Bottom_Beacon</object-name>
  </animation>

  <animation>
    <name>StaticGrouped</name>
    <noshadow />
    <object-name>Left_Wing_Static_Red</object-name>
    <object-name>Right_Wing_Static_Green</object-name>
    <object-name>Tail_Static_White</object-name>
  </animation>

  <animation>
    <name>StrobeGrouped</name>
    <noshadow />
    <object-name>Left_Wing_Strobe</object-name>
    <object-name>Right_Wing_Strobe</object-name>
  </animation>
```
hope it helps...

**Comment by wlbragg** [comment]
@Megaf I changed the title on this because, like I said, I am working on Compositor lighting and we might as well add your issue to that entire issue. I will address it in this issue anyway. Now we have the start of a Compositor lighting PR complete with a per-existing bug :)

**Comment by wlbragg** [comment]
Notes to me:
Courtesy, no MP alias

**Comment by wlbragg** [comment]
Can anyone tell me if the c172p courtesy lights are on the dimer rheostat that dims the white cabin dome light? I know the courtesy is on the dome light switch, but I don't know if they are always full on or supposed to also be tied to the dimer rheostat. We had them full on  but I don't know if that was an oversight or by design.

**Comment by wlbragg** [comment]
OK, first round done. Needs some serious testing. Clustered shading lighting is now working for...
Landing Light
Taxi Light
Courtesy Lights
Nav Lights
Strobe Lights
Beacon Light
Everyone watching this thread, please test.

**Comment by legoboyvdlp** [comment]
Thank you! I will try it :)

**Comment by legoboyvdlp** [comment]
![image](https://user-images.githubusercontent.com/10434911/103794128-458c1400-503c-11eb-869b-95292abbb4cd.png)


It's looking reasonably good. Is the white light at the rear

**Comment by legoboyvdlp** [comment]
Sorry -- hit wrong button.

**Comment by wlbragg** [comment]
>Is the white light at the rear causing illumination? It looks even dimmer than the nav lights.

It's pointed in a different direction than the other two nav's. It also has slightly different settings.

>And the landing lights don't seem to illuminate grass, unlike the strobes

Hum, that doesn't sound right, I'll check that out.

>That's a C172M. Unless, its a LED or something like that. But, I think its much brighter than the one we have.
>In addition, maybe the beacon should be brighter:

I'll try adjusting some of this. Also @HHS81 made some really cool code for the beacon that changes the blink to a slower faded effect. I really like it. I just hadn't done that pert yet as it required some nasal changes in the electrical system.

Thank you for reviewing!

**Comment by wlbragg** [comment]
@legoboyvdlp I'm going to do a second round of adjustments. Right now your seeing a combination of ALS and the new clustered-shading lighting. For example the beacon is a new point light and the rest are all spot lights with and ALS procedural light. I'm not sure why @HHS81 did it exactly this way? Maybe you have to to get the best results? Maybe he isn't finished? Maybe I missed something (I'm seeing code in my sleep). The beacon is a point light, I think the rest of the lights with the exception of landing and taxi lights need to be point lights as well.
I am going to try to do that and see what it looks like.
Next comes the interior. I really want to see what can be accomplished with these new lights. It may turn out that we need a combination of both light maps and these new lights.
Part of my reasoning for using the new lighting scheme instead of ALS is that it works for the default renderer and ALS as well. Overall we should be able to reduce the complexity of the lighting.

**Comment by wlbragg** [comment]
Well, I guess I didn't totally understand what the clustered-shading light are capable of. It looks like they are only for illuminating the objects and have no object like even the ALS procedural lights.
Don't we need an ALS procedural type light for the default rendering under Compositor?

**Comment by wlbragg** [comment]
@HHS81 I think you will want to change all your lights except landing and taxi to point. For example, if you look at the image above of the aircraft with the strobes caught on camera, you can see they appear illuminate the fuselage ,I think the nav will to, at least a little. I think they are more a directional point light, not a spot.

**Comment by wlbragg** [comment]
@HHS81 ah, more noise on my part. Looks like the point lights are not blocked by objects and appear on the opposite side of a two sided object. I guess that's related to your questioning for shadows! So maybe that is why it has to be a spot light. Although the nav's and strobes almost look like they could be points.

**Comment by wlbragg** [comment]
Look at this one, kind of hard to see but don't you think the NAV light illuminated the wing tip at least a foot or two in towards the fuselage?
![Navigation-Lighting-Market](https://user-images.githubusercontent.com/11069680/103850520-4a60c000-505c-11eb-8995-08c7af04fa17.jpg)

**Comment by wlbragg** [comment]
@legoboyvdlp I changed some things around. When you get a chance, please check again. The only thing I still want to do is adopt @HHS81 beacon nasal code which will give the appearance of a brighter beacon light.

**Comment by legoboyvdlp** [comment]
> Well, I guess I didn't totally understand what the clustered-shading light are capable of. It looks like they are only for illuminating the objects and have no object like even the ALS procedural lights.
> Don't we need an ALS procedural type light for the default rendering under Compositor?

Correct. The lights only produce illumination of the model and scenery. The best thing to do is to keep the ALS procedural lights (+ legacy for non-ALS?) but remove the lightspots + ALS secondary lights. The good thing is that the clustered shading lights work on all renderers now.

**Comment by HHS81** [comment]
@wlbragg 
> 
> 
> @legoboyvdlp I'm going to do a second round of adjustments. Right now your seeing a combination of ALS and the new clustered-shading lighting. For example the beacon is a new point light and the rest are all spot lights with and ALS procedural light. I'm not sure why @HHS81 did it exactly this way? Maybe you have to to get the best results? Maybe I missed something (I'm seeing code in my sleep). The beacon is a point light, I think the rest of the lights with the exception of landing and taxi lights need to be point lights as well.
> I am going to try to do that and see what it looks like.
> Next comes the interior. I really want to see what can be accomplished with these new lights. It may turn out that we need a combination of both light maps and these new lights.
> Part of my reasoning for using the new lighting scheme instead of ALS is that it works for the default renderer and ALS as well. Overall we should be able to reduce the complexity of the lighting.

> 
> 
> Look at this one, kind of hard to see but don't you think the NAV lights illuminate the wing tip at least a foot or two in towards the fuselage?...)

What you can see on your image is a reflection of the lights. Something we would need raytracing or something similar for:
![light example](https://user-images.githubusercontent.com/12649375/103914436-1fcd4080-510a-11eb-89d2-ce88f843f0a8.jpg)

If you would use a pointlight, everything 360° around this pointlight would illuminate.
A pointlight acts like a candle or light bulb; a spotlight like a flashlight/ torch. 

(Maybe a cheap solution to simulate the picture above would be to put a pointlight at the location where the ground will reflect the light and animate the AGL position of this pointlight and their light power depending on AGL.)

I decided to use the lights like in real: Landing/taxilights are spotlights. Navlights and wingstrobe has specified angles so spotlights are the best for as well. The beacon on the C182S seems more a to be a point light, so I used a pointlight here. On older aircraft it is a rotating beacon, so an animated spotlight for the beacon would be correct. 

Btw.: the lightnings of storms should also use the new lights. Maybe it is even possible that the new lights will illuminate clouds and fog?

**Comment by HHS81** [comment]
@wlbragg  https://ppltutor.com/air-law/rules/aircraft_lights/

**Comment by wlbragg** [comment]
A little something to compare to except this is LED.
https://www.youtube.com/watch?v=DDWD81JJwxE

@legoboyvdlp I am finished. I completely removed ALS exterior lighting including the landing and taxi, except the procedural stuff. All clustered rendering is illuminating the grass now.
I adjusted several areas of the external lighting including positions and strengths as well as timing of the strobes and beacon.

@HHS81 take a look at my changes if you get the chance. I think you'll like some of it, maybe even better than the c182 implementation, other parts, maybe not. I think this looks good and fairly accurately represents what I see in comparison images and videos. Note the fading of the nav lights illumination on the ground VS the hard edges the c182 is showing. By the way, you can eliminate those hard edges without adopting the point lights if you want. It's just a matter of adjusting.

**Comment by HHS81** [comment]
hmm... in the video the strobes seems not to illuminate the interior or the rest of the aircraft. That would be bad, as they would blind the pilot. That´s the reason why in clouds and fog they should be switched off according Torsten Dreyer. 
Here is another video showing where the led-strobes are aiming: https://www.youtube.com/watch?v=1VUy61OW8v8

But your nav lights looks correct with those soft edges. Yes, that´s something  will adopt.  Thanks a lot!

**Comment by legoboyvdlp** [comment]
We actually don't need raytracing since flightgear can render the illumination of lights on aircraft now I think :)

**Comment by wlbragg** [comment]
I saw that saw that video. That is partially what I was basing some of my observations on. What may be fooling me is the white reflective hanger walls bouncing light into the aircraft fuselage. Because you can definitely see the white strobe illuminating the doors of the fuselage. But is that the reflection from the hanger only?

**Comment by legoboyvdlp** [comment]
The strobes look good.
![image](https://user-images.githubusercontent.com/10434911/103947457-fd4e1e00-512f-11eb-8791-99b72b89af00.png)

Is there any documentation on any of the wing lights (nav / strobe), if they are directional at all?

**Comment by legoboyvdlp** [comment]
I think the beacon could be brighter still, same for the navlights.
![image](https://user-images.githubusercontent.com/10434911/103947539-22db2780-5130-11eb-9341-bbcfdb4a9bb7.png)


But, on the whole this is good, and perhaps photos aren't so reliable anyway as they might have external lighting or be overexposed :)

**Comment by HHS81** [comment]
> 
> 
> The strobes look good.
> 
> Is there any documentation on any of the wing lights (nav / strobe), if they are directional at all?

Yep, there are regulations which angle the lights have to illuminate. The strobes shouldn`t illuminate directly the interior

**Comment by HHS81** [comment]
> 
> 
>  Because you can definitely see the white strobe illuminating the doors of the fuselage. But is that the reflection from the hanger only?

Yes, I think so. Just look closer on such wingtip strobes and how they are built up. You can see that there should no light directly light the doors, or cabin

**Comment by HHS81** [comment]
> 
> 
> We actually don't need raytracing since flightgear can render the illumination of lights on aircraft now I think :)

Raytracing is much more than just lights :-)

**Comment by HHS81** [comment]
Hmm- how about that:
![fgfs-20210107214833](https://user-images.githubusercontent.com/12649375/103949365-6d15d600-513b-11eb-886d-82f84a84584d.png)
![fgfs-20210107214844](https://user-images.githubusercontent.com/12649375/103949369-6f783000-513b-11eb-95f4-a2caaca55373.png)

**Comment by wlbragg** [comment]
How cool is that!

**Comment by HHS81** [comment]
I´m not sure if it is too bright now

**Comment by HHS81** [comment]
@wlbragg 
I like your landing and taxilights a lot! With the right atmosphere and scenery everything looks really nice!
![fgfs-20210107232415](https://user-images.githubusercontent.com/12649375/103956228-66419000-5148-11eb-9a67-69f113f31e72.jpg)
![fgfs-20210107232508](https://user-images.githubusercontent.com/12649375/103956237-693c8080-5148-11eb-9e78-c5c2fc2cab0f.jpg)

**Comment by wlbragg** [comment]
>I think the beacon could be brighter still, same for the navlights.

If so, are you talking about the points of light themselves at the wingtip and the beacon light or the illumination of the ground (in the case of the nav lights) and tail (in the case of the beacon)

>perhaps photos aren't so reliable anyway as they might have external lighting or be overexposed 
Yes, I wonder the same thing. I remember back when we set the procedural lights up there was much discussion and I remember toning the ALS nav light spots on the ground down as the consensus was they should be barely visible in broad daylight and subdued at night.

Check out some of these nighttime videos of the cabin of aircraft in flight
https://www.youtube.com/watch?v=dJh2Ror81N0 Look close at the wing strut (beacon illumination) and the cabin interior wall where the glass mounts (slight strobe or beacon illumination?) @ 2:25

And this one...
https://www.youtube.com/watch?v=nRYNoi1vL9U
The beacon is clearly visible on the wing strut and a couple times in the grainy video portion you can see the view out the back and there is a lot of light in that view (not necessarily in the cabin though). Towards the end when on the ground the exterior gets quit actively lit by strobes and beacon. There is one sort portion where you can see the strobes reflection in the passengers headset.

So I agree, I need to try to tone the strobes down to where they really don't influence the interior of the aircraft.

**Comment by wlbragg** [comment]
This is about all I can do, I keep gong back and forth, so I guess I'll leave it here.

**Comment by legoboyvdlp** [comment]
That looks brilliant, thanks :+1:

**Comment by Megaf** [comment]
> @Megaf I changed the title on this because, like I said, I am working on Compositor lighting and we might as well add your issue to that entire issue. I will address it in this issue anyway. Now we have the start of a Compositor lighting PR complete with a per-existing bug :)
> 
> I may have something pushed to a new branch for this later today and it should include the fix for the strobe shadows. I imagine they never needed a noshadow flag before but do now.
> 
> What we will all find exciting is that that shadow is going to turn into light illumination on the fuselage instead.

@wlbragg That's not a problem, I just gave a try to branch 1349 (The one with Compositor lights) and it is looking real good.
Looking forward to seeing this merged.

The only thing I see is that when Compositor lights are off then there are no lights at all. Not sure if actually an issue with the c172p as I understand there's not much that can be done about it since in FG the "generic" shader is obligatory for lights.

The reason why one would disable generic is to gain more FPS and responsiveness out of the Sim.

**Comment by wlbragg** [comment]
I think  my reasoning was that ALS is not an option anymore, it's part of the default rendering (so to speak) and so from this point forward it is no longer necessary to support lights for the lowest settings.
Maybe I need to rethink this or get some kind of consensus. But lighting is so cheap now it seems like we don't need to accommodate this any longer?
But maybe I made a mistake about this and we need the most basic lights if shaders are completely off. I also don't remember exactly what state I left the interior panel and post lighting in under a "no shader" setting.

**Comment by wkitty42** [comment]
FWIW: i think i would target this new lighting stuff to the compositor flavor of FG and not worry about backward compatibility, TBH... set the minimum FG version to 2020.4.0 for now and adjust it to whatever 2020.4 becomes when it comes out... t'would make things a lot easier to manage and the team can concentrate on the new stuff and not have to worry about older FG at all...

**Comment by wlbragg** [comment]
@legoboyvdlp could you take a look at this and give your opinion as to what else needs to be supported if anything from your perspective?

**Comment by legoboyvdlp** [comment]
Sure, no problem! Are you looking specifically towards backwards compatibility or anything in general?

**Comment by wlbragg** [comment]
Yes, also I never really looked at interior lighting effect and if we need to remove any backwards compatibility. Although that really hasn't diverged as of yet, I guess?

**Comment by legoboyvdlp** [comment]
The good news is that the Compositor lights just don't affect earlier versions, so we can have them :) 

The bad news is that we probably have to add a version check (<= 2020.3) to the ALS secondary lights and the lightspots. It shouldn't be too bad though.

**Comment by legoboyvdlp** [comment]
Boosted the courtesy lights a little. Judging from this image, they are relatively bright.
![image](https://user-images.githubusercontent.com/10434911/107652284-556cc880-6c78-11eb-8640-be0fa628ea15.png)


![image](https://user-images.githubusercontent.com/10434911/107652256-4c7bf700-6c78-11eb-8c6e-af5afe022671.png)


Also added a dome light: what do you think? 
![image](https://user-images.githubusercontent.com/10434911/107653599-b8129400-6c79-11eb-9066-e9b7bf92718a.png)

I think it complements the existing lightmaps well. 

Otherwise the interior light is probably fine - the lightmaps do a fantastic job.


**Point lights**

There is a formula which is often used in OpenGL rendering to get plausible attenuation values for point lights. In short, linear = 4.5 / (range) and quadratic = 75 / (range)^2.

Applying this formula and reducing the range of the beacon light, resulted in:
![image](https://user-images.githubusercontent.com/10434911/107655151-302d8980-6c7b-11eb-8ede-f51ce05835ab.png)

The navlights were certainly too dim: I think this is a better overall result. 

![image](https://user-images.githubusercontent.com/10434911/107656255-e2fde780-6c7b-11eb-8385-9bdceedad270.png)


Looking at a real photo, the beacon light might even need to be brightened, but I don't really mind. Do feel free to comment and adjust anything!
![image](https://user-images.githubusercontent.com/10434911/107656384-032da680-6c7c-11eb-9962-701307ae0ccb.png)

**Comment by legoboyvdlp** [comment]
Backwards compatibility:

1) We need to make sure that 2020.3 users are not negatively affected.
2) On 2020.4, there's no problem with compatibility. Exterior lights are visible at all quality levels, as are lightmaps. 

Since we already have a Cessna in the LTS, I am happy to go ahead with the assumption that from now on the Cessna in the repository will be for 2020.4 or later. 

I don't know the status of the Cessna and what happened for 2020.3 as I wasn't monitoring things at that point. But if we do that, we should cut a release/2020.3 branch now, and make sure to put any bugfixes onto that branch. 

Does this make sense? That avoids the complexity of adding many <conditions> and enabling / disabling lights depending on renderer.

**Comment by wlbragg** [comment]
Replying piecemeal, as I'm on my cell phone away from a big screen and nice images. Firslt, regarding ALS secondary lights and light spots, they don't exist anymore that's the new compositor lights,  they're completely gone? So I thought.
Regarding dome light, I can't see it real well on my phone. Did you create a compositor style dome lamp lumination in addition to the light maps? If so, I will take a close look at that when I get to my computer. I tried to do that on the air crane, and could not get the cut off correct and it always bled through to the outside of the aircraft if you look closely.
Any visual settings you changed, increased/decreased, I'm sure I'll be fine with, but I will take a closer look when I get to my computer and can pull this.

**Comment by legoboyvdlp** [comment]
Many thanks for the reply 👍 

I've simply hidden the dome outside with a select animation. It's simple but works 😉

**Comment by legoboyvdlp** [comment]
Just noting to self: 
* the electrical power source of lights
* brightness of the lightmap which is controlled by the radio light switch (it seems to be the illumination from the backlight of the radio and engine instruments. 
*

**Comment by wlbragg** [comment]
@legoboyvdlp
OK, I had a good look. I have some issues that I discovered along the way that maybe you aren't aware of or our monitors are rendering light different than each other.

First let me say I don't disagree with you that the nav lights could possibly stand to be a little brighter. But I was under the opinion it is the point light portion that needed the increase. Increasing the compositor light is problematic. If you increase them to the extent of your changes we start to get light in places where it shouldn't be, on the underside of objects that should be in shadow and also we start getting hard edges, which in my opinion don't look good and take away from the realism.After thinking about it I am almost inclined to use the compositor light only to illuminate the ground and scene objects such as buildings and trees and then make light maps for the body objects of the aircraft. They are so much more accurate and realistic. I don't know if you adjusted the position of the nav lights at all, but I actually had them offset from where the source is in all cases to both help eliminate that line and to keep the light from illuminating the underside of the wing and underside of the fuselage and elevators, which I don't think it should be illuminating. Again, maybe our monitors are displaying the light different from one another.

See the first image with the critique of the hard edges and the illumination of the underside of the wings. You can see the hard edge from both the read nav and both wings.
![light1](https://user-images.githubusercontent.com/11069680/107854919-23707900-6dd4-11eb-8a34-82338382cda8.jpg)

Here it is without the beacon and my edits which shows it better.
![light2](https://user-images.githubusercontent.com/11069680/107854927-32efc200-6dd4-11eb-81f4-8c85a06031d2.jpg)

.Here from a different angle
![light3](https://user-images.githubusercontent.com/11069680/107854935-3b47fd00-6dd4-11eb-8e6e-43ec3fd8a23f.jpg)

Here it is before the changes and with what I felt was the correct "absence of light" on the underside. Keep in mind, maybe our monitors are showing this differently.
![image4 jpg](https://user-images.githubusercontent.com/11069680/107855395-d4781300-6dd6-11eb-9112-7dd2db7d5645.png)

Here is the difference between the two when looking at the ground reflection, I really don't think we should see that hard edge on the ground either. Looking at the most illuminated part (the center) it looks like maybe you didn't reposition the nav light.
![light](https://user-images.githubusercontent.com/11069680/107855509-87e10780-6dd7-11eb-8f15-7b3f68e05295.gif)

Per your picture the courtesy lights are pretty bright.and I agree they could be a bit brighter.

Now the interior dome compositor, no I don't like it. The light maps are far more accurate as they are pre-rendered ray traced maps. The compositor lighting is casting light through solid objects to the shaded side as you noticed when you had to place the interior view select animation so to eliminate the bleed through of the dome light to the underbelly of the exterior fuselage. It's the light maps that are still keeping the interior looking decent because they contain the shadows. All the compositor lighting is doing is illuminating the shadow. I think it is distracting from the scene, in my  opinion. But, if it is the shade of light, or the amount that your not happy with, we can adjust both the color of the light maps and the intensity.
![light](https://user-images.githubusercontent.com/11069680/107855584-2a00ef80-6dd8-11eb-8692-c89e7e4f9459.gif)

So, I guess I am disagreeing with some of this, your thoughts?

**Comment by legoboyvdlp** [comment]
That's fine with me, the dome light was only an experiment!

I think you may be right about the monitor; those look much much harder than they were here. 

> After thinking about it I am almost inclined to use the compositor light only to illuminate the ground and scene objects such as buildings and trees and then make light maps for the body objects of the aircraft. 
I don't believe that's possible; the lights (as you've noticed) illuminate irrespective of circumstances.

> I don't know if you adjusted the position of the nav lights at all
Not that I know of! :) 

> , but I actually had them offset from where the source is in all cases to both help eliminate that line and to keep the light from illuminating the underside of the wing and underside of the fuselage and elevators, which I don't think it should be illuminating

That makes sense. Keep in mind that I did adjust attenuation to match the formula provided by openGL --> so perhaps this is some of it. Let me try reducing the range and so on, to see if it can get any better? 

This doesn't look right : you can't see the rear nav light, but it _is_ illuminating forwards. Maybe it should be a spot rather than a point in this case.
![image](https://user-images.githubusercontent.com/10434911/107856093-cf8a8200-6e1d-11eb-9357-8ad3dc29828e.png)



Do you have any comment on the backwards compatibility?

**Comment by legoboyvdlp** [comment]
I've edited the beacon such that it doesn't illuminate the bottom of the aircraft, at least on my side.

This is a navlight variant: how does this look?
![image](https://user-images.githubusercontent.com/10434911/107856684-b1268580-6e21-11eb-8e05-980b2ca719e7.png)


Strobes:
![image](https://user-images.githubusercontent.com/10434911/107856694-c6031900-6e21-11eb-96dd-db4169a86162.png)


If you want to revert back to the original state, that's totally fine too :)

**Comment by wlbragg** [comment]
@legoboyvdlp that is way better. I have only two possible issues. The red and green nav lights (compositor) I think maybe are a bit to bright on the ground. All the other issues with them are gone. I'm not sure how you got them adjusted so well, but they look really good. I'm probably going to use these setting for the PA-18 and the AirCrane, as far as intensity (softened a little). I'm just not sure they should be quite that bright. I'd really like to get some other opinions. The white tail is great, I wouldn't change it at all. The beacon and strobes look great with one exception about the beacon. Look at the trailing back half at the top of the wings and the fuselage when it flashes. It has that hard line, if you can do something to soften and get rid of that line, it would be perfect.
So my only two questions is brightness of red and green nav light on the ground and the hard edge of the beacon on the top of the wings.
I still have to look at the backwards compatibility again before I can comment.

**Comment by wlbragg** [comment]
Interesting, /sim/rendering/shaders/skydome is what we use to drive the backwards compatibility modeling. Exterior lighting seems to have no change whatsoever when turning the model shader all the way down in the rendering GUI. You no longer can change /sim/rendering/shaders/skydome in the GUI, nore does it change at startup in the launcher if you switch the rendering to "Default". So should we eliminate all non-ALS exterior lighting. Did the model shader slider ever have any effect on the exterior lighting. Or was it only /sim/rendering/shaders/skydome that was effecting that? Using the QT GUI rendering switch, what is the difference now between Default and ALS rendering?

If I choose"Default" rendering in the QT GUI and turn the sim model shader slider all the way down, I am getting a black interior. What is that all about? @legoboyvdlp can you confirm this?

**Comment by wlbragg** [comment]
Note to myself: we use /sim/rendering/shaders/skydome condition in many places in the code.
Verify Effects\shadow\ files are not yet removed in any branch or PR.
Depending on the above, either remove shadow code in an existing PR or make a new issue and PR. All other compositor related code changes are included in either this issue PR or fg1000, I think.

**Comment by Megaf** [comment]
Gentlemen, so, for lights skydome (ALS) should not matter at all.
Since Compositor supports lights "natively". That said, Compositor seems to be backwards compatible with ALS.

My suggestion is, remove from the code all checks for skydome, and only check if Generic shader is enabled. The model shader isn't related anymore.

**Comment by legoboyvdlp** [comment]
Hasn't that property /sim/rendering/sky-dome been removed?

I don't have a Default option any more, so I can't really test that. The only option right now is Compositor.

ALS and shaders depends on the quality levels and is always on. That brings a question to mind, what happens with custom shader settings.

I think removing the specific model shader references would be a good idea.

>So should we eliminate all non-ALS exterior lighting.
I'll confirm this but I'm pretty sure that for all quality levels, ALS lighting is now available.

Thanks for your comments; I'll work on the navlights this afternoon!

**Comment by wlbragg** [comment]
@legoboyvdlp @Megaf I'm still not clear about the settings.

>Hasn't that property /sim/rendering/sky-dome been removed?No I think it is still there, just maybe not changed any longer.

Right, but I am getting some really bad effects. I wouldn't change anything until we discuss farther. I'm going through a couple different version of the aircraft and different setting to try to determine what is going on.

@legoboyvdlp  please check interior with the model shader set below 2, using both default and ALS rendering in the QT Launcher, as you know ALS is no longer in the in sim GUI, but it is still in the property tree. It does seem to no longer be used at all though. Setting QT rendering to Default VS ALS does not change that property. So what is is changing? IS it changing the model shader settings or something else, or is it changing nothing?

@legoboyvdlp @Megaf keep in mind /sim/rendering/sky-dome is being used extensively throughout our aircraft code. It's also not just for our lights, I'm not sure what all it controls in the sim rendering.
What I really want to know is what is model shader < 2 changing? I know it's tied to techniques in the effects files. Why is our aircraft interior effect broke with model shader < 2?
@legoboyvdlp can you verify this? I get the same breakage using fgdata c172pc version.

Once we figure out what is wrong, if /sim/rendering/sky-dome is not ever being changed anymore, we can remove al the conditional code using it.

![int1](https://user-images.githubusercontent.com/11069680/107884838-ee375a00-6eab-11eb-8140-61c4c4c34f9f.jpg)

![int2](https://user-images.githubusercontent.com/11069680/107884840-f1cae100-6eab-11eb-8375-758d6afb8837.jpg)

![int3](https://user-images.githubusercontent.com/11069680/107884843-f42d3b00-6eab-11eb-93c2-4fd3739381af.jpg)

**Comment by wlbragg** [comment]
There is also this, I'm not sure how long these lines have been here. Someone needs to test with a FG version prior to the major compositor handoff. Is anyone following using LTS? Everything I'm showing is on next as of few days ago. If this is truly broke, I need to report this to the devs.

![int4](https://user-images.githubusercontent.com/11069680/107885053-d6140a80-6eac-11eb-9b81-147fcccfa9d3.jpg)

**Comment by wkitty42** [comment]
On 2/14/21 7:54 AM, legoboyvdlp wrote:
> Hasn't that property /sim/rendering/sky-dome been removed?

the option to turn it on/off has been removed but it is still there because so 
many craft use it and haven't been updated to work without it...

**Comment by wlbragg** [comment]
@wkitty42 what version are you using and do you get the black interior with model shader set below 2?

**Comment by wlbragg** [comment]
I changed the heading, we can do the /sim/rendering/sky-dome changes and remove the ALS 3d shadows here, once we're sure all these statements about /sim/rendering/sky-dome are true. We have to be positive!

**Comment by legoboyvdlp** [comment]
Per Stuart, 
```3) Remove the /sim/rendering/shaders/skydome predicate from all the ALS fgdata/Effects
4) In defaults.xml set /sim/rendering/shaders/skydome=true.  This will ensure that the Aircraft effects with predicates that reference /sim/rendering/shaders/skydome will work as before.```

Stuart subsequently confirmed that Erik had done the above, and then said:

```I've just pushed a fix to next that sets /sim/rendering/shaders/skydome based on the quality level:  Quality level 0 or 1 sets /sim/rendering/shaders/skydome=0, while quality levels 2 set /sim/rendering/shaders/skydome=1.  This simply extends the existing code that sets the various shader parameters based on the quality level.
... a very basic low quality level with ALS switched off, and ALS switched on for higher quality levels.```



This means that we should continue to use /sim/rendering/shaders/skydome I think.
But that is not going to work, I need to send an email to the mailing list. This is a FlightGear bug.

Can you tell me what the value of /sim/rendering/shaders/quality-level-internal is please?

> Setting QT rendering to Default VS ALS does not change that property. 

How are you setting the QT rendering setting? It doesn't exist anymore?

**Comment by legoboyvdlp** [comment]
I can't reproduce the black interior effect.

Sorry, re the above,
 I see icecode brought up this issue already.

Does this issue by any chance go away, if you unselect custom shaders and set the quality level directly to `5` and then go back to custom shaders?

**Comment by wkitty42** [comment]
On 2/14/21 2:14 PM, wlbragg wrote:
> @wkitty42 what version are you using

next generally updated ASAP when i see new commits going into code via the 
commit logs mailing list...

> and do you get the black interior with model shader set below 2?
unknown... am in the middle of a rather long flight, ATM... will try to remember 
to test a little later... remind me, please, do i need to select a specific 
branch of the dev craft for this test? i think i do...

**Comment by wlbragg** [comment]
@wkitty42 

>remind me, please, do i need to select a specific
branch of the dev craft for this test? i think i do...

It would be helpful if you could test fgdata version and https://github.com/c172p-team/c172p/tree/Issue-1349 on the dev branch. What FG versions would you be able to test on and if next, from approximately what date?

@legoboyvdlp what version of FG did you test for the black interior textures on? I will check what you asked in a bit, as soon as I am done with  some shuttle work I am doing at the moment. To much to do and to little time to do it.

**Comment by wkitty42** [comment]
On 2/14/21 4:51 PM, wlbragg wrote:
> @wkitty42
> 
>     remind me, please, do i need to select a specific
>     branch of the dev craft for this test? i think i do...
> 
> It would be helpful if you could test fgdata version and 
> https://github.com/c172p-team/c172p/tree/Issue-1349 on the dev branch.

ok...

> What FG versions would you be able to test on and if next, from approximately
> what date?
i am always updating my next to the severely sharp bleeding edge ASAP... 
sometimes that's several times a day... compiling takes maybe a minute for my 
system unless OSG gets an update to 3.6... then it may add another 2 or 3 
minutes unless it is a really deep change that affects a lot of the OSG code...

**Comment by wlbragg** [comment]
Well if we find the black texture is a bug and you can reproduce it on next, then I might ask you to try to locate when it broke.

**Comment by wlbragg** [comment]
>Does this issue by any chance go away, if you unselect custom shaders and set the quality level directly to 5 and then go back to custom shaders?

@legoboyvdlp 
Wow, I'm not sure what the heck just happened...
No custom, quality set to 2 - 5 = good (0-5 stages allowed, 6 stages)
Back to Custom, set model to 0 = bad (0-3 stages allowed, 4 stages)
or
No custom, quality set to 0 - 1 = good (0-5 stages allowed, 6 stages)
Back to Custom, set model to 0 or 1 = good (0-1 stages allowed, 2 stages)

So yeah, I guess it's buggy. Does this fit what you heard from @icecode?

**Comment by legoboyvdlp** [comment]
> @legoboyvdlp what version of FG did you test for the black interior textures on? I will check what you asked in a bit, as soon as I am done with some shuttle work I am doing at the moment. To much to do and to little time to do it.

2020.4 always. I'm on -next. 

I think you might not be on -next, because there isn't a default / ALS / rembrandt selector anymore; it's entirely gone? 😕

**Comment by legoboyvdlp** [comment]
![image](https://user-images.githubusercontent.com/10434911/107948381-a687ff80-6f8b-11eb-9072-a70d100e3858.png)


![image](https://user-images.githubusercontent.com/10434911/107948431-b9023900-6f8b-11eb-9e3f-fb1b8d7ccead.png)

**Comment by legoboyvdlp** [comment]
@wlbragg are you still seeing this problem?

**Comment by wlbragg** [comment]
@legoboyvdlp sorry, I've been unavailable for a few days. I'll rebuild tonight or tomorrow and try to help finish this us.

**Comment by wlbragg** [comment]
@legoboyvdlp I just wiped my build folders and pulled pulled and merged current next and I still have the Default/ALS selector, no Rembrandt.

Also I am still seeing the black interior if I turn the model slider all the way down in custom mode.

/sim/version/flightgear: 2020.4.0
/sim/version/simgear: 2020.4.0
/sim/version/openscenegraph: 3.4.2
/sim/version/build-id: none
/sim/version/build-number: 0
/sim/version/build-type: Dev
/sim/version/revision: 9e77cdf6cbfa89554e626d28477eba2d863bba50
/sim/rendering/gl-vendor: ATI Technologies Inc.
/sim/rendering/gl-renderer: Radeon (TM) RX 480 Graphics
/sim/rendering/gl-version: 4.6.14757 Compatibility Profile Context 20.12.1 27.20.14501.28009
/sim/rendering/gl-shading-language-version: 4.60
/sim/rendering/max-texture-size: 16384
/sim/rendering/depth-buffer-bits: 24

**Comment by wkitty42** [comment]
On 2/24/21 5:29 PM, wlbragg wrote:
> [...]  and I still have the Default/ALS selector,
> no Rembrandt.

that's how it should be... not only in the launcher but also in-sim...

**Comment by legoboyvdlp** [comment]
But that's the whole point. As I've been saying you shouldn't have _any_ selector, not even for Default or ALS. It simply doesn't exist any more.

 I think there is something wrong ... do double check you are using the correct launcher?

![image](https://user-images.githubusercontent.com/10434911/109154274-f9b93980-7765-11eb-9354-910e1d328705.png)

**Comment by wlbragg** [comment]
OK, so where does the QT launcher get picked up from? I thought clearing the flightgear-build directory should compile a completely new version, including QT?

**Comment by legoboyvdlp** [comment]
Yes compiling FlightGear should be sufficient. I suspect you aren't calling the correct fgfs.exe but have an older one. It depends of course if you deleted the old .exe.

1. If you have an /install/ folder make sure you clear that too. It shouldn't be necessary but just in case.
2. Check you don't have compilation errors which prevent the new .exe from being copied to overwrite the old one.

**Comment by wlbragg** [comment]
It was the windows-3rd-party. I now have the correct QT launcher profile. In the sim there is still the custom checkbox. When selected I get steps on the model slider when not selected you only get the performance slider. If I have custom selected and I set the model slider to 0 - 1 I still get the black interior.

**Comment by legoboyvdlp** [comment]
Awesome, at least that's ruled out.  That all sounds normal now! :) 

Does this happen on the non-compositor branch too? If so perhaps this should really go to the mailing list.

**Comment by wlbragg** [comment]
Yes, it even happens on fgdata version. I am starting the discussion on the mailing list.

In the meantime, I just happened to get into the sim on a bright sunny day and the red and green nav light are to bright on the runway in broad daylight, noticeably so, at least on my monitor. If you get a chance maybe you can tone it down, I think in daylight it should be really subtle, hopefully that will still allow it to be bright enough at night. If we can't get both day and night to our liking then I think the sim rendering of light intensity is suspect. If that is the case, then we either have to artificially adjust brightness against something like /rendering/scene/diffuse/red or I would rather have a little lower intensity that wanted at night than the bright intensity on the runway in the daylight, it looks so bad to me when it is so bright in the day. If you disagree, please say so.

**Comment by legoboyvdlp** [comment]
Will do!

**Comment by wlbragg** [comment]
@legoboyvdlp per the title of this issue, I am going to continue to remove features that are no longer needed since compositor came online. Like all the shadow code.
I want to get this issue and associated PR done. As soon as it is done, maybe we can set a new version and make it available for fgdata.

I fixed the "black" interior issue. It was something in my fgdata, I pulled a clean copy vs updating the existing one and it fixed it.

Speaking of backwards compatibility, I'm still unclear as to what we want to do on this aircraft. On the rest of my projects I am removing all backwards compatibility of shadow code and lighting.
Beings this is the only aircraft shipping with FG I want to make sure we are doing what is expected from the majority or at least the key core members. It is still totally unclear to me where the separation between backwards compatibility and current rendering is.

I'm going to go through all the interior and exterior lighting and the ?model shader settings and see what is there and exactly how it is affected now that there is no distinction between ALS and default rendering.

I'll try to post here a legend of what is there and what settings affect it. If you already know these questions and answers and can articulate it to me, that'd be even better.

**Comment by wlbragg** [comment]
Well that was actually not to difficult to figure out once I got my shader issue figured out.
We have some complicated lighting modeled.
Interior lighting...
First the basics are
1) lightmaps (ALS)
require /sim/rendering/shaders/skydome = true
model shader =  2 or 3
Everything that was ALS rendering used these lightmaps for everything except there are 2 gauges that also had Integral lighting, for that we used a material animation if post lighting was off.

2) material animations (basic rendering)
/sim/rendering/shaders/skydome = 0
model shader = 0 or 1

I guess if we want backwards compatibility still with model shader set to 0 or 1 we need to simply remove the /sim/rendering/shaders/skydome = 0 condition and replace it with model shader < 2.

Make sense?

Exterior lighting...
All the lighting is tied to /sim/rendering/shaders/skydome and the model shader slider has no effect. So I assume it is safe to remove all the backwards compatibility for the exterior lighting. Everyone should be able to see the existing lighting in this branch regardless of settings, right?

Ground shadows...

The only thing about ground shadows that I question is that there is a choice in the in sim GUI under View\Rendering Options to turn off shadows. That is the compositor based shadows that choice is referencing. Is there any conceivable reason to leave the old 3d shadow option? Or should shadow choice only be compositor or nothing?

**Comment by legoboyvdlp** [comment]
> Beings this is the only aircraft shipping with FG I want to make sure we are doing what is expected from the majority or at least the key core members. It is still totally unclear to me where the separation between backwards compatibility and current rendering is.


Rembrandt is certainly completely gone. I think your solution makes sense.


I don't mind leaving the old shadow; just whatever you prefer!

I'll get the navlights adjusted later on today.

**Comment by wlbragg** [comment]
I'm going to put this to dev mail list, thanks for the feedback.

**Comment by legoboyvdlp** [comment]
![image](https://user-images.githubusercontent.com/10434911/109399881-676b8e00-793d-11eb-83ab-2599c1faeffe.png)

![image](https://user-images.githubusercontent.com/10434911/109399887-74887d00-793d-11eb-95c5-e1adca74e01f.png)

![image](https://user-images.githubusercontent.com/10434911/109399894-7e11e500-793d-11eb-9026-1c5b46e63c20.png)

![image](https://user-images.githubusercontent.com/10434911/109399937-c4ffda80-793d-11eb-838a-0aaef86491f9.png)

Does this look ok?

**Comment by wlbragg** [comment]
Ya, much better in the daylight. You OK with it at night?

As soon as I get a couple more replies (or not) on the mailing list I will push my changes to the lighting and shadows. We can then test a bit and maybe we can get this merged.

**Comment by legoboyvdlp** [comment]
OK! That's great. 

Yep that seems ok to me.

**Comment by Megaf** [comment]
Just one little thing, the navigation lights, they look like a spot light pointed at the ground.

They are very bright and quickly fade, when they should fade over a longer distance. I don't even know if it's possible to do that though. 

Also,  are the lights respecting the angles for navigation lights? They are only supposed to be seen at certain angles.

**Comment by legoboyvdlp** [comment]
Need firm data, sorry, beyond "certain angles" 😅 

Yes thats exactly what the lights are, they are spot lights. They need to be spotlights as otherwise they will illuminate the wing and fuselage?

**Comment by wkitty42** [comment]
somewhere there are specs for the lights that should provide the viewing angles...

**Comment by Megaf** [comment]
![image](https://user-images.githubusercontent.com/6201512/109426442-a5bc8800-79e5-11eb-8ca6-bfdd943c1936.png)

**Comment by Megaf** [comment]
One more thing, I couldn't find a single c172p with landing/taxi lights on the winds, they all have them on the nose just bellow the propeller.
![image](https://user-images.githubusercontent.com/6201512/109426798-57a88400-79e7-11eb-828a-20adcdecd331.png)
![image](https://user-images.githubusercontent.com/6201512/109426806-5bd4a180-79e7-11eb-97af-9832f7096e64.png)
![image](https://user-images.githubusercontent.com/6201512/109426815-6000bf00-79e7-11eb-9051-2c45dfcdc0ca.png)
![image](https://user-images.githubusercontent.com/6201512/109426817-64c57300-79e7-11eb-8a11-c416108804f0.png)

**Comment by wkitty42** [comment]
On 2/28/21 12:07 PM, Megaf wrote:
> One more thing, I couldn't find a single c172p with landing/taxi lights on the 
> winds, they all have them on the nose just bellow the propeller.

this model is from 1982, IIRC... it is also based on a IRL craft that has moved 
them from the nose to the wing... the actual craft is one of the liveries...

zoom in on this image and you can see the light on the left wing...
https://kcdn.hangar67.com/photos/11265/51e372e641.jpg

and this one, too...
https://media.sandhills.com/n62507-1982-cessna-172p-skyhawk/img.axd?id=77476225&wid=6072144879&rwl=False&p=&ext=&w=614&h=460&t=&lp=&c=True&wt=False&sz=Max&rt=0&checksum=dC%2faonMu6Xz99xd4efLovo5hK9bPZXXRgNwbv%2fyiD8Y%3d

**Comment by HHS81** [comment]
> ---edit---
> Ok, I found actually 2 or 3 photos of C172P with the lights on the wing root...

Yes, there are, and have been considered when the project started years ago! It is the model of the C172 P  later version, when landing and taxilights changed their position.

Btw. Nice to see the D-EDDB now finally flying after such a long bulding time..... 🤣

**Comment by HHS81** [comment]
> 
> Just one little thing, the navigation lights, they look like a spot light pointed at the ground. They are very bright and quickly fade, when they should fade over a longer distance. I don't even know if it's possible to do that though. Also, are the lights respecting the angles for navigation lights? They are only supposed to be seen at certain angles.

As much I could follow here, they are alreay have the correct angles, and what looks like spots is because real position lights are spots: spotlights which emits light only at certain angles. 

Just wait until the developers are ready with their work!

**Comment by wlbragg** [comment]
The lights have always been modeled to represent the real life characteristics as far as viewing angle. 
The clusterd-shading compositor lighting is done in my opinion.

The forward view (viewing from the front) of the procedural lighting is slightly off. According to the image above the light should fade at 90 and are are reaching to around 130.

![range](https://user-images.githubusercontent.com/11069680/109429239-87d43080-79af-11eb-847a-b520e288ae79.jpg)

@legoboyvdlp you feel like having a crack at this?

The yellow is where our lighting is cutoff compared to where it should be.

The files are in Effects\lights\procedural-light-nav-xxx.eff
The pertinent code is...
~~~
        <!-- Angles are 0.0 at 0 deg from pointing direction, 1.0 at
             90/-90 deg, and 0.0 at 180/-180 deg.

             For left navigation light we use -0/-35 .. +110/+145 for
             the inner/outer range. This gives an arc of 110/180 deg,
             or 55/90 deg from center.

             Value = sin(angle in degrees)

             0.8191520442889918 = 55 deg (* 2 = 110 deg inner angle)
             1.0000 = 90 deg (* 2 = 180 deg outer angle)
        -->
        <inner_angle type="float">0.8191520442889918</inner_angle>
        <outer_angle type="float">1.0</outer_angle>
        <zero_angle type="float">0.982547593563</zero_angle>
        <outer_gain type="float">0.1</outer_gain>
~~~

I'm currently removing 3d shadows and backwards compatible exterior lighting and correcting the interior lighting backwards compatibility. 

It's pretty close already, is the c182s using the same techniques? If so are your procedural nav lights cutoff more precisely? That'd make it real easy to fix, :)

**Comment by wlbragg** [comment]
Looking back at something @HHS81 posted
https://ppltutor.com/air-law/rules/collision_lights/
If the forward cutoff was truly at the 90, you would technically see either the red or green nav light when head on?
So maybe they need to stay where they are?

**Comment by legoboyvdlp** [comment]
> Also, due to the lights location, it wouldn't matter how bright they are, they would never illuminate the wings and aircraft body.

That's unfortunately not relevant; because FlightGear's would.

**Comment by legoboyvdlp** [comment]
@wlbragg It does seem that the procedural lights have a mathematical error causing this. I've improved it (previously it seemed that it was 180 degrees, now its a max of 110 degrees). 

I suspect this is because of this:
 Angles are 0.0 at 0 deg from pointing direction, 1.0 at
             90/-90 deg, and 0.0 at 180/-180 deg.

In that scheme its not possible to differentiate between sin(125) and sin(55) for instance. Both of them are the same value.

I've fixed the tail light though. Now it cuts off at 140 degrees as it should.

**Comment by HHS81** [comment]
@wlbragg 
> 
> 
> Looking back at something @HHS81 posted
> https://ppltutor.com/air-law/rules/collision_lights/
> If the forward cutoff was truly at the 90, you would technically not see either the red or green nav light when head on, there would be a blind spot?
> So maybe they need to stay where they are?

![aircraft_lights](https://user-images.githubusercontent.com/12649375/109430929-03f86380-7a04-11eb-8853-617070e9dee6.jpg)

in the top middle of the image I used the viewing angles provided by FlightGear in sin/views/current-view. If you look straight to the front of the aircraft you will see both navlights. If you are a bit off (about 1°) you will only see one of them, it is cut off.
I have implemented it correctly for the ALS procedural lights and the new compositor lights.

**Comment by wlbragg** [comment]
@legoboyvdlp 
OK, thanks. I think it's good enough. The nav lights work for forward collision as they are, that's all you ask!
One more thing, as I am removing the exterior lighting backward compatibility, I noticed the red beacon seems to be wrong again, or I never noticed it. It is lighting up the underside of the fuselage and you can see a hard edge on the ground when illuminated.
I would swear we had that fixed and it was soft and where we wanted it. It's really noticeable on my monitor.

**Comment by legoboyvdlp** [comment]
Ok ... I'll try and get it back to that 😁

**Comment by wlbragg** [comment]
Sorry, thanks!

**Comment by legoboyvdlp** [comment]
> in the top middle of the image I used the viewing angles provided by FlightGear in sin/views/current-view. 

That's an idea. I will try that.

**Comment by HHS81** [comment]
> 
> 
> > in the top middle of the image I used the viewing angles provided by FlightGear in sin/views/current-view.
> 
> That's an idea. I will try that.

I don`t think it will work- doesn`t use the model space system. procedural als lights and compositor uses model space system (as in Blender)

**Comment by wlbragg** [comment]
@legoboyvdlp 
I just check the c182s and the nav lights forward angle is really close to what ours is. Actually it might be identical, so I am good with it.

You need to take a look at their beacon, the light being cast is really nice. It is bright and cuts-off well. I assume that portion is just the compositor lighting. Maybe it will give you some ideas for the settings. The actual beacon is some custom code that is also really nice, but I don't know if we should "steal" it just yet! :)

**Comment by Megaf** [comment]
> @legoboyvdlp
> I just check the c182s and the nav lights forward angle is really close to what ours is. Actually it might be identical, so I am good with what we have.
> 
> You need to take a look at their beacon, the light being cast is really nice. It is bright and cuts-off well. I assume that portion is just the compositor lighting. Maybe it will give you some ideas for the settings. The actual beacon is some custom code that is also really nice, but I don't know if we should "steal" it just yet! :)

You'd be just borrowing that, it's ok!
I think the interior lights are spot on already! And I agree with removing the old stuff and keeping just compositor stuff.
The light angles was my only reservation and everything else looks pretty good the way it is right now.

**Comment by wlbragg** [comment]
@legoboyvdlp
OK, I pushed a big update.

Removed 3d shadows and all associated supporting code. I commented out a couple lines referring to MP shadow support until I am sure it is no longer needed.

# Disable ALS 3D shadow for remote aircraft because of issue #387
 #rplayer.initNode("sim/rendering/shadow-volume", 0, "BOOL");

Will we want to keep this for backwards compatibility when running under MP or is it no longer appropriate?

I changed all the lighting select animations to use model shader property VS skydome. I think it works as expected.
I forgot to condition the compositor left and right nav lighting on wing damage as well as it just being powered on, so I fixed that.

I will remove a bunch of code that I currently only have commented out as soon as I feel comfortable of these changes.

As far as I am concerned, after testing and fixing the beacon we should be ready to merge this to master.

I was also thinking, before we do merge into master we may want to tag a final pre-compositor release.

**Comment by wkitty42** [comment]
AFAIK, MP shadow stuff should not be needed with compositor builds because compositor does shadows for everything as needed...

i would also be sure to set the minimum FG version to 2020.3(.6?) or if we're further along, 2020.4(.0?) to be sure that older FG cannot use the craft... especially since, IIRC, backwards compatibility with pre-compositor is being eliminated...

**Comment by legoboyvdlp** [comment]
OK - I'll check out the beacon and make sure that all looks ok with various settings.

And yes it should be set to 2020.4.0. 2020.3.6 Compositor is not 2020.4.0 Compositor, there are differences and the former should be considered a deprecated preview.

**Comment by legoboyvdlp** [comment]
@wlbragg do you know where I can find the Compositor c182s? I searched the code using Notepad++ Find and replace for <light> and couldn't see any of them. 

this is what I see :
![image](https://user-images.githubusercontent.com/10434911/109504657-b3304b80-7a93-11eb-9dbc-8ac93812ee76.png)

**Comment by wlbragg** [comment]
@legoboyvdlp 
My c182s shows as Issue358

**Comment by legoboyvdlp** [comment]
Thanks!

**Comment by legoboyvdlp** [comment]
![image](https://user-images.githubusercontent.com/10434911/109673858-581f5700-7b6e-11eb-9701-b08718611d3e.png)

This is what I see with the same settings as c182s. Note the hard edges and it also goes in underneath the aircraft.

I think it will need a bit more work

**Comment by legoboyvdlp** [comment]
Right, I've tried making it into a spotlight. Not sure about this, what do you think? I think its better than the point light although certainly not perfectly.

![image](https://user-images.githubusercontent.com/10434911/109676620-cb29cd00-7b70-11eb-8983-7ac82375a24b.png)

**Comment by wlbragg** [comment]
@legoboyvdlp 
I'm not sure how you did it, but that looks great. Really nice!

OK, I want to build one more time with Fernando's new rendering layout and make sure our lighting implementation still work as we expect it to. After that and we test things a bit, we can get this merged.
I'm still thinking we should make a final pre-compositor release and then merge this. What do you think? 2020.1 is the last tagged release I think.

**Comment by wkitty42** [comment]
what's been done between the last tagging and now that is fixes and not compositor related changes?

did the new 8.3 radio stuff get merged in?

the new kap140?

i'd only worry about those if getting a final tag/release before compositor is necessary/desired... plus whatever bug fixes, of course... if this stuff is already in and done, then just tag 2020.3.7 and get it in place for that coming up in a week or two... after that, unless there's more bug fixes needed, i'd leave it alone and just do the compositor changes with an eye on tagging that as 2021.something...

**Comment by legoboyvdlp** [comment]
Tagging a 2020.3 release would be good to get the last state before a newer 2020.4 / 2021.1 is required.
